pycbc/detector.py

```diff
@@ -201,6 +206,31 @@ def set_gmst_reference(self):
             raise RuntimeError("Can't get accurate sidereal time without GPS "
                                "reference time!")
 
+    def lal(self):
+        """ Return lal data type detector instance """
+        if hasattr(self, '_frDetector'):
+            return self._frDetector
+        else:
+            import lal
+            d = lal.FrDetector()
+            d.vertexLongitudeRadians = self.longitude
+            d.vertexLatitudeRadians = self.latitude
+            d.vertexElevation = self.info['height']
+            d.xArmAzimuthRadians = self.info['xangle']
+            d.yArmAzimuthRadians = self.info['yangle']
+            d.xArmAltitudeRadians = self.info['yaltitude']
+            d.xArmAltitudeRadians = self.info['xaltitude']
+
+            # This is somewhat abused by lalsimulation at the moment
+            # to determine a filter kernel size. We set this only so that
+            # value gets a similar number of samples as other detectors
+            # it is used for nothing else
+            d.yArmMidpoint = 4000.0
+
+            x = lal.Detector()
+            r = lal.CreateDetector(x, d, lal.LALDETECTORTYPE_IFODIFF)
```

Comment:
Couldn't we just set this to self._frDetector here before returning (or have the if be `if not hasattr` do this and then always just return self._frDetector.

Reply:
Yep, already pushed this change.